src/block/interface.ts

```diff
@@ -0,0 +1,72 @@
+/* eslint-disable @typescript-eslint/no-unnecessary-type-constraint */
+/* eslint-disable no-use-before-define */
+import { Link, Version } from '../link/interface.js'
+import { CID } from "../cid.js"
+
+/**
+ * A byte-encoded representation of some type of `Data`.
+ *
+ * A `ByteView` is essentially a `Uint8Array` that's been "tagged" with
+ * a `Data` type parameter indicating the type of encoded data.
+ *
+ * For example, a `ByteView<{ hello: "world" }>` is a `Uint8Array` containing a
+ * binary representation of a `{hello: "world"}.
+ */
+export interface ByteView<Data> extends Uint8Array, Phantom<Data> {}
+
+/**
+ * A utility type to retain an unused type parameter `T`.
+ * Similar to [phantom type parameters in Rust](https://doc.rust-lang.org/rust-by-example/generics/phantom.html).
+ *
+ * Capturing unused type parameters allows us to define "nominal types," which
+ * TypeScript does not natively support. Nominal types in turn allow us to capture
+ * semantics not represented in the actual type structure, without requring us to define
+ * new classes or pay additional runtime costs.
+ *
+ * For a concrete example, see {@link ByteView}, which extends the `Uint8Array` type to capture
+ * type information about the structure of the data encoded into the array.
+ */
+export interface Phantom<T> {
+  // This field can not be represented because field name is non-existings
+  // unique symbol. But given that field is optional any object will valid
```

Comment:
```suggestion
  // This field can not be represented because field name is nonexistent
  // unique symbol. But given that field is optional any object will valid
```